dialog/file.go

```diff
@@ -164,9 +174,11 @@ func (f *fileDialog) refreshDir(dir string) {
 
 		itemPath := filepath.Join(dir, file.Name())
 		if file.IsDir() {
-			icons = append(icons, f.newFileItem(theme.FolderIcon(), itemPath))
+			icons = append(icons, f.newFileItem(canvas.NewImageFromResource(theme.FolderIcon()),
+				itemPath, file.Name(), true))
 		} else {
-			icons = append(icons, f.newFileItem(theme.FileIcon(), itemPath))
+			icons = append(icons, f.newFileItem(NewFileIcon(itemPath), itemPath,
```

Comment:
Why does the constructor take an icon derived from the path, a name derived form the path, and the path itself?
It doesn't seem right that the filedialog needs helpers to setup data to pass into a constructor